GOVERNANCE.md

```diff
@@ -8,7 +8,7 @@ related projects which include:
 
 - [nodejs/iojs.org](https://github.com/nodejs/iojs.org) published to https://iojs.org
 - [nodejs/nodejs.org](https://github.com/nodejs/iojs.org) published to https://nodejs.org
-- [joyent/node-website](https://github.com/joyent/node-website) published to https://nodejs.org
+- [joyent/node-website](https://github.com/joyent/node-website) 
```

Comment:
Seems fine to just link. Mind updating it to point at https://github.com/nodejs/nodejs.org-archive?